### DISCUSSION TITLE
自定义vue节点的连接桩固定在左上角，无法像默认节点自动计算位置，求解

### DISCUSSION THREAD

**Discussion by ColumbusK** [antvis/X6#4325] [Q&A]
![mmexport1716960720982.png](https://github.com/antvis/X6/assets/60766772/a47d5d18-d051-445f-8d5b-4b3be62045cd)

我大概知道是和connecting 中的 anchor 锚有关，但具体不知道如何解决

1. 是在注册自定义节点时配置吗？
2. 还是说需要自己实现，比如节点组件监听位置变化，进行连线的向量运算重新计算自定义节点的锚点位置

**Comment by ColumbusK** [comment]
已解决
检查元素才意识到，在addNode添加节点时，必须指定w h，否则节点容器大小为1✖️1

**Comment by Tolerate-yu** [comment]
> ![mmexport1716960720982.png](https://private-user-images.githubusercontent.com/60766772/334665558-a47d5d18-d051-445f-8d5b-4b3be62045cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3NDUyNjEsIm5iZiI6MTcyNDc0NDk2MSwicGF0aCI6Ii82MDc2Njc3Mi8zMzQ2NjU1NTgtYTQ3ZDVkMTgtZDA1MS00NDVmLThkNWItNGIzYmU2MjA0NWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDA3NDkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwMzE0ODM1ODUyNTU3OThjMjNjNjMyMGU5MWJkMDRiNWY2MWY3MzVhMjVjYjgwMjQyY2M4M2EwM2FmOTg4MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lmMV1nssMI4cDN-yxw3NrJkYbMKjRwoHFdKPeGwnVRc)
> 
> 我大概知道是和connecting 中的 anchor 锚有关，但具体不知道如何解决
> 
> 1. 是在注册自定义节点时配置吗？
> 2. 还是说需要自己实现，比如节点组件监听位置变化，进行连线的向量运算重新计算自定义节点的锚点位置

这个设置连接桩了嘛？
